src/operator/controllers/pod_reconcilers/pods.go

```diff
@@ -73,17 +74,6 @@ func (p *PodWatcher) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 		return ctrl.Result{}, errors.Wrap(err)
 	}
 
-	// If a new pod starts, check if we need to do something for it.
-	var intents otterizev1alpha3.ClientIntentsList
```

Comment:
Can we add a check in: `addOtterizePodLabels` to see if the pod's service identity is a target of any ClientIntents, or a client of any ClientIntents, and only then label it?